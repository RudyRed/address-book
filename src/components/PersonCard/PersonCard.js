import { Paper, Grid, Typography } from '@material-ui/core';

const PersonCard = ({ firstName, lastName, phoneNumber }) => {
  return (
    <Paper elevation={4}>
      <Grid
        container
        direction="column"
        alignItems="center"
        justifyContent="center"
      >
        <Grid item>
          <Typography variant="h2">{firstName}</Typography>
        </Grid>
        <Grid item>
          <Typography variant="h2">{lastName}</Typography>
        </Grid>
        <Grid item>
          <Typography variant="h2">{phoneNumber}</Typography>
        </Grid>
      </Grid>
    </Paper>
  );
};

export default PersonCard;
