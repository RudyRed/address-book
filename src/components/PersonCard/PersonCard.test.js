import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

import PersonCard from './PersonCard';

jest.mock('@material-ui/core', () => ({
  ...jest.requireActual('@material-ui/core'),
  Avatar: (props) => <div {...props} />,
}));

const defaultProps = {
  firstName: 'PersonCard-firstName',
  lastName: 'PersonCard-lastName',
  phoneNumber: 'PersonCard-phoneNumber',
};

const setup = (overrides) => {
  const props = {
    ...defaultProps,
    ...overrides,
  };
  return {
    ...render(
      <Router>
        <PersonCard {...props} />,
      </Router>,
    ),
    props,
  };
};

describe('PersonCard', () => {
  it('Should render PersonCard', () => {
    const { getByText, props } = setup();

    expect(getByText(props.firstName)).toBeInTheDocument();
    expect(getByText(props.lastName)).toBeInTheDocument();
    expect(getByText(props.phoneNumber)).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });
});
