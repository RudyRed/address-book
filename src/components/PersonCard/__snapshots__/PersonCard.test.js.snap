// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonCard matches snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation4 MuiPaper-rounded"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2"
        >
          PersonCard-firstName
        </h2>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2"
        >
          PersonCard-lastName
        </h2>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2"
        >
          PersonCard-phoneNumber
        </h2>
      </div>
    </div>
  </div>
  ,
</div>
`;
